extensions/mongodb-client/deployment/src/test/java/io/quarkus/mongodb/MongoMetricsTest.java

```diff
@@ -0,0 +1,65 @@
+package io.quarkus.mongodb;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.eclipse.microprofile.metrics.Counter;
+import org.eclipse.microprofile.metrics.MetricID;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.eclipse.microprofile.metrics.Tag;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClient;
+
+import io.quarkus.mongodb.metrics.MongoMetricsConnectionPoolListener;
+import io.smallrye.metrics.MetricRegistries;
+
+public class MongoMetricsTest extends MongoTestBase {
+
+    private MongoClient client;
+    private MetricRegistry metricRegistry;
+
+    @BeforeEach
+    void init() {
+        client = com.mongodb.client.MongoClients
+                .create(MongoClientSettings.builder().applyConnectionString(new ConnectionString(getConnectionString()))
+                        .applyToConnectionPoolSettings(
+                                builder -> builder.addConnectionPoolListener(new MongoMetricsConnectionPoolListener()))
```

Comment:
IIUC this connection pool listener is needed for updating the metrics, but here you only explictly add it in a test, but how will it be added in production code? I suppose that when `quarkus.mongodb.metrics.enabled` is true, then we need to do this automatically. So something like this needs to be in the place in Quarkus codebase that produces Mongo clients (`AbstractMongoClientProducer` I think?) To be able to do that, `AbstractMongoClientProducer` will need to be able to see the value of `quarkus.mongodb.metrics.enabled` property, so I think it might need to be turned into a runtime property